### PR TITLE
Adding kuenm, ellipsenm, and grinnell 

### PR DESCRIPTION
**Pull request by marlonecobos**
kuenm:
- occ_cleaning:  exploring records outside layer borders included
- env_collinearity: PCA analyses implemented
- backg_sample: options to sample background included
- data_partitioning: option for partitioning data
- mod_fit: fits maxent models
- mod_tuning: model calibration based on feature classes, regularization multipliers, and sets of variables
- mod_ensemble: multiple selected models are used to create consensus
- mod_evaluate: evaluation of models based on distinct metrics
- pred_general: predictions to other scenarios or areas are possible with functions
- pred_extrapolation: analyses of extrapolation risks using MOP
- pred_inspect: variable importance can be explored
- post-processing: other post-processing (post-modeling) analyses allowed, variability analyses

ellipsenm:
- occ_cleaning: remove NA and duplicates 
- env_collinearity: plots to explore variable correlation, PCA analyses implemented
- bias: thinning and exploring records outside layer borders included
- study_region: options to create calibration areas included
- backg_sample: options to sample background included
- data_partitioning: two options for partitioning data
- mod_fit: fits distinct types of ellipsoids to data
- mod_tuning: model calibration based on ellipsoid type and sets of variables
- mod_evaluate: evaluation of models based on distinct metrics
- pred_general: predictions to other scenarios or areas are possible with functions

grinnell:
- env_collinearity: PCA analyses implemented
- study_region: options to simulate calibration areas are the main focus
- backg_sample: options to sample background included
- mod_fit: fits ellipsoids to data
- pred_general: predictions to other scenarios or areas are possible with functions
- post-processing: dispersal simulations using multiple parameters and inputs are allowed

### CODE REVIEW

File: inst/extdata/packages/ellipsenm.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+niche and species distribution modeling, this package is the first in
+presenting a comprehensive set of tools that use ellipsoid envelopes as
+the theoretical representation of ecological niches. Details on the
+theoretical background and methods used here can be found in:
+Peterson et al. (1999) <doi:10.1126/science.285.5431.1265>,
+Soberon and Peterson (2005) <doi:10.17161/bi.v2i0.4>,
+Peterson (2011) <doi:10.1111/j.1365-2699.2010.02456.x>,
+Peterson et al. (2011) <https://bit.ly/2UpQ9Hd>."
+occ_acquisition: no
+occ_cleaning: yes
```
Comment (jamiemkass) [inline inst/extdata/packages/ellipsenm.yaml:24]: Not clear what function "exploring records outside layer borders included" refers to. If it means `to_closest`, I suppose that can qualify as cleaning.

File: inst/extdata/packages/kuenm.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Peterson et al. (2011) <https://www.degruyter.com/princetonup/view/title/506966>,
+Barve et al. (2011) <doi:10.1016/j.ecolmodel.2011.02.011>,
+Owens et al. (2013) <doi:10.1016/j.ecolmodel.2013.04.011>,
+Radosavljevic and Anderson (2014) <doi:10.1111/jbi.12227>,
+Peterson et al. (2028) <doi:10.1111/nyas.13873>,
+Cobos et al. (2019a) <doi:10.7717/peerj.6281>,
+Cobos et al. (2019b) <doi:10.1016/j.ecoinf.2019.100983>,
+Cobos et al. (2019c) <doi:10.1101/603100>."
+occ_acquisition: no
+occ_cleaning: yes
```
Comment (jamiemkass) [inline inst/extdata/packages/kuenm.yaml]: Not clear what function "exploring records outside layer borders included" refers to. If it means to_closest, I suppose that can qualify as cleaning.

File: inst/extdata/packages/kuenm.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+data_integration: no
+env_collinearity: yes
+env_process: no
+bias: no
+study_region: no
+backg_sample: yes
+data_partitioning: yes
```
Comment (jamiemkass) [inline inst/extdata/packages/kuenm.yaml]: Which function does this? I couldn't find it.

File: inst/extdata/packages/kuenm.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+mod_fit: yes
+mod_tuning: yes
+mod_ensemble: yes
```
Comment (jamiemkass) [inline inst/extdata/packages/kuenm.yaml]: I also couldn't find the ensemble function here.

File: inst/extdata/packages/ellipsenm.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+data_integration: no
+env_collinearity: yes
+env_process: no
+bias: yes
```
Comment (jamiemkass) [inline inst/extdata/packages/ellipsenm.yaml]: Please list the function that does this.

File: inst/extdata/packages/ellipsenm.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+mod_tuning: yes
+mod_ensemble: no
+mod_stack: no
+mod_evaluate: yes
+mod_multispecies: no
+mod_mechanistic: no
+pred_general: yes
+pred_extrapolation: no
+pred_inspect: no
+post_processing: no
```
Comment (jamiemkass) [inline inst/extdata/packages/ellipsenm.yaml]: Doesn't this pkg do niche overlap analyses? That should qualify for post-processing.

File: inst/extdata/packages/grinnell.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
+name: ellipsenm
```
Comment (jamiemkass) [inline inst/extdata/packages/grinnell.yaml]: Please change these fields to those for `grinnell`.

**Review by jamiemkass — CHANGES_REQUESTED** [review]
Some changes need to be made, but looks good. Please make sure to fix the name, title, etc. for `grinnell`.

**Comment by sylvainschmitt** [comment]
Thanks @jamiemkass for the review.

**Comment by sylvainschmitt** [comment]
@marlonecobos  , your contribution can be integrated without further edits. After merging your pull request, you will be added to the repository contributors. Please expect to be contacted for a review request in the future.

**Pull request merged**